### DISCUSSION TITLE
Errors with groovy version and with OSM data

### DISCUSSION THREAD

**Discussion by cathguil** [orbisgis/geoclimate#744] [Q&A] [answered]
Hello, 

I need to use the Geoclimate plugin in order to map the LCZ on a particular city. 

I'm working on Windows with OSM data and I helped myself with the tutorials available on github (using the command prompt and groovy) but I have two problems:

- on the command prompt, I get this error message :
![image](https://user-images.githubusercontent.com/95683820/173603519-aba9b008-1c49-4d8d-8d1d-4a9b56fbd6cf.png)

- on groovy, I get this error message
![image](https://user-images.githubusercontent.com/95683820/173603408-efd37045-48b3-4529-bd4d-acc5197ab2e4.png)

Do you know where these errors come from and how to solve them please?  

Thank you in advance,

**Comment by ebocher** [comment, marked as answer]
As noted in the error message, you must used Groovy 3.0.9 and it seems you have installed the 3.0.7 version.

**Reply by cathguil** [reply]
Thank you very much for your answer. But now I have another problem using groovy 3.0.9 version : 
![image](https://user-images.githubusercontent.com/95683820/173609900-6e478133-f8b9-4535-ad53-c07091dec0fc.png)

Can you help me please ?

**Reply by j3r3m1** [reply]
It seems there is trouble inside your configuration file. Can you share it with us ?

**Reply by cathguil** [reply]
Yes, here is my configuration file : 

![image](https://user-images.githubusercontent.com/95683820/173769469-99a4c9ea-15d0-488d-9576-5a801d9cb201.png)

I also tried to change "locations" with "osm" in the input but the problem is still the same.

**Reply by cathguil** [reply]
Hello, 
I would like to come back to you following the message I posted about twenty days ago. I would like to know if you have found the source of my problem to use Geoclimate plugin. 
I thank you in advance for your feedback.

**Reply by j3r3m1** [reply]
Hi @cathguil , sorry for the late answer.
Can you copy and paste the content of your config file instead of an image ?

**Reply by cathguil** [reply]
Yes, the content of my config file is the one below : 

{
    "description": "Processing OSM data",
    "input": {
        "osm": [
            "Perpignan"
        ]
    },
    "output": {
        "folder": "C:\Users\guill\mydirectory\Geoclimate"
    },
    "parameters": {
        "rsu_indicators": {
            "indicatorUse": [
                "LCZ",
                "TEB",
                "UTRF"
            ],
            "svfSimplified": true,
            "estimateHeight": true
        },
        "grid_indicators": {
            "x_size": 100,
	    "y_size": 100,
	    "rowCol": false,
	    "output" : "geojson",
	    "indicators" :[
                 "BUILDING_FRACTION", 
                 "BUILDING_HEIGHT", 
                 "WATER_FRACTION",
                 "VEGETATION_FRACTION", 
                 "ROAD_FRACTION", 
                 "IMPERVIOUS_FRACTION", 
                 "LCZ_FRACTION"
             ]
         }
    }
}

And this is the error message I get : 

C:\Users\guill\mydirectory\Geoclimate>java -jar Geoclimate.jar -f my_first_config_file_osm.json -w OSM
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
1 osm areas will be processed
https://nominatim.openstreetmap.org/search?q=Perpignan&limit=5&format=geojson&polygon_geojson=1
Executing query... Perpignan
Downloading the Nominatim data.
Error while executing the process.
groovy.lang.MissingPropertyException: Exception evaluating property 'size' for java.util.ArrayList, Reason: groovy.lang.MissingPropertyException: No such property: size for class: java.math.BigDecimal

**Comment by j3r3m1** [comment]
OK I am not sure I understand. You now have a different error than previously. Your previous error ("the parameter input has no value") was related to your config file which I suppose was wrong. Now the new error you have is related to the use of the BDTopo data using the Windows OS. 
I am not sure this bug has been fixed yet. @ebocher ?
